### DISCUSSION TITLE
How to cancel an event stream using htmx sse extension and avoid automatic reconnection

### DISCUSSION THREAD

**Discussion by bnixon67** [bigskysoftware/htmx#2144] [Q&A]
I have been exploring the htmx sse extension and appreciate the ease of use and functionality that it provides.

I am unsure how to cancel an event stream. Per https://web.dev/articles/eventsource-basics, it says "To cancel a stream from the server, respond with a non text/event-stream Content-Type or return an HTTP status other than 200 OK (e.g. 404 Not Found)."

When I do this, the stream stops, but then the event stream is re-initiated, potentially due to the Automatic Reconnection logic?

How can I avoid automatic reconnection in this case?

**Comment by soypat** [comment]
```js
document.body.addEventListener('htmx:sseClose', function(e) {
document.body.removeAttribute('sse-connect');
});
document.body.addEventListener('htmx:sseError', function(e) {
document.body.removeAttribute('sse-connect');
});
```
